go/vt/vttablet/tabletserver/debugenv.go

```diff
@@ -70,90 +70,132 @@ func debugEnvHandler(tsv *TabletServer, w http.ResponseWriter, r *http.Request)
-			setIntValCtx(tsv.SetPoolSize)
-		case "StreamPoolSize":
-			setIntValCtx(tsv.SetStreamPoolSize)
-		case "TxPoolSize":
-			setIntValCtx(tsv.SetTxPoolSize)
-		case "MaxResultSize":
-			setIntVal(tsv.SetMaxResultSize)
-		case "WarnResultSize":
-			setIntVal(tsv.SetWarnResultSize)
-		case "RowStreamerMaxInnoDBTrxHistLen":
-			setInt64Val(func(val int64) { tsv.Config().RowStreamer.MaxInnoDBTrxHistLen = val })
-		case "RowStreamerMaxMySQLReplLagSecs":
-			setInt64Val(func(val int64) { tsv.Config().RowStreamer.MaxMySQLReplLagSecs = val })
-		case "UnhealthyThreshold":
-			setDurationVal(func(d time.Duration) { tsv.Config().Healthcheck.UnhealthyThreshold = d })
-			setDurationVal(tsv.hs.SetUnhealthyThreshold)
-			setDurationVal(tsv.sm.SetUnhealthyThreshold)
-		case "ThrottleMetricThreshold":
-			setFloat64Val(tsv.SetThrottleMetricThreshold)
-		case "Consolidator":
-			tsv.SetConsolidatorMode(value)
-			msg = fmt.Sprintf("Setting %v to: %v", varname, value)
+		f(durationVal)
+		msg = fmt.Sprintf("Setting %v to: %v", varname, value)
+		return nil
+	}
+
+	setFloat64Val := func(f func(float64)) error {
+		fval, err := strconv.ParseFloat(value, 64)
+		if err != nil {
+			return fmt.Errorf("invalid float64 value for %v: %v", varname, err)
 		}
+		f(fval)
+		msg = fmt.Sprintf("Setting %v to: %v", varname, value)
+		return nil
+	}
+
+	var err error
+	switch varname {
+	case "ReadPoolSize":
+		err = setIntValCtx(tsv.SetPoolSize)
+	case "StreamPoolSize":
+		err = setIntValCtx(tsv.SetStreamPoolSize)
+	case "TxPoolSize":
+		err = setIntValCtx(tsv.SetTxPoolSize)
+	case "MaxResultSize":
+		err = setIntVal(tsv.SetMaxResultSize)
+	case "WarnResultSize":
+		err = setIntVal(tsv.SetWarnResultSize)
+	case "RowStreamerMaxInnoDBTrxHistLen":
+		err = setInt64Val(func(val int64) { tsv.Config().RowStreamer.MaxInnoDBTrxHistLen = val })
+	case "RowStreamerMaxMySQLReplLagSecs":
+		err = setInt64Val(func(val int64) { tsv.Config().RowStreamer.MaxMySQLReplLagSecs = val })
+	case "UnhealthyThreshold":
+		err = setDurationVal(func(d time.Duration) { tsv.Config().Healthcheck.UnhealthyThreshold = d })
+	case "ThrottleMetricThreshold":
+		err = setFloat64Val(tsv.SetThrottleMetricThreshold)
+	case "Consolidator":
+		tsv.SetConsolidatorMode(value)
+		msg = fmt.Sprintf("Setting %v to: %v", varname, value)
+	}
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
 	}
 
+	vars := getVars(tsv)
+	sendResponse(r, w, vars, msg)
+}
+
+func handleGet(tsv *TabletServer, w http.ResponseWriter, r *http.Request) {
+	vars := getVars(tsv)
+	sendResponse(r, w, vars, "")
+}
+
+func sendResponse(r *http.Request, w http.ResponseWriter, vars []envValue, msg string) {
+	format := r.FormValue("format")
+	if format == "json" {
+		respondWithJSON(w, vars, msg)
+		return
+	}
+
+	respondWithHTML(w, vars, msg)
+}
+
+func getVars(tsv *TabletServer) []envValue {
 	var vars []envValue
-	vars = addVar(vars, "PoolSize", tsv.PoolSize)
+	vars = addVar(vars, "ReadPoolSize", tsv.PoolSize)
```

Comment:
Is this change not backward incompatible? Don't we need to keep the old and new name around?

Reply:
I do not think it is important to keep it backward compatible for debug/env vars.
But, I do not have strong opinions about it.